records/generator.go

```diff
@@ -448,6 +484,9 @@ func slaveIdTail(slaveID string) string {
 // zk://username:password@host1:port1,host2:port2,.../path
 // file:///path/to/file (where file contains one of the above)
 func getProto(pair string) (string, string, error) {
```

Comment:
instead of `SplitN` we should probably attempt to `net.SplitHostPort` -- and if that fails we could to parse as a URL. Either way, I think that work should be done in a separate PR.